omnimatter/prototypes/recipes/extraction-dynamic.lua

```diff
@@ -267,7 +267,22 @@ for i, tier in pairs(omnisource) do
 		setMain(item):
 		setLevel(3*omni.pure_levels_per_tier):
 		setEnergy(function(levels,grade) return 5*(math.floor((grade-1+(tier_int-1)/2)/levels)+1) end):
-		setTechIcon(ore.mod or "omnimatter","extraction-"..item):
+		setTechIcon(function(levels, grade)
+			local ore_icon = omni.icon.of(item, "item")
+			for _, layer in pairs(ore_icon) do
+				layer.shift = {
+					0,
+					layer.icon_size * 0.5 * (64 / layer.icon_size)
+				}
+			end
+			return omni.lib.add_overlay(
+				{{
+					icon = "__omnimatter__/graphics/technology/extraction-generic.png",
+					icon_size = 128
+				}},
+				ore_icon
+			)
+		end):
```

Comment:
Please move functions that require more than 1 line out of RecGen, hard to read otherwise